pkg/optimizely/cache.go

```diff
@@ -114,6 +119,11 @@ func (c *OptlyCache) UpdateConfigs(sdkKey string) {
 	}
 }
 
+// SetUserProfileService sets maps userProfileService against the given sdkKey
+func (c *OptlyCache) SetUserProfileService(sdkKey, userProfileService string) {
```

Comment:
Please add lock here.

Reply:
we are using `userProfileServiceMap cmap.ConcurrentMap` here which handles concurrency itself.

Reply:
I think we should still add the mutex here. For reading the ConcurrentMap should be fine but when setting, I believe the mutex should be used to ensure no race condition, even with the ConcurrentMap, just to be safe.. I could be wrong though.